src/core/memory.d

```diff
@@ -1024,6 +1025,26 @@ void __delete(T)(ref T x) @system
     }
 }
 
+/// ditto
+void __delete(T)(ref const(T)[] x) @system
+if (is(const(T) : T))
+{
+    import core.internal.traits : Unqual;
+    auto x2 = cast(Unqual!T[]) x;
```

Comment:
Why not doing this in the mutable overload?

Reply:
Fair point!